samples: cmsis_rtos_v2: Add sample application (#11648)
zephyrproject-rtos/zephyr: samples/portability/cmsis_rtos_v2/README.rst

.. _cmsis_rtos_v2-sample:

> Dining Philosophers

Reviewer: For the table of contents, since there's already a sample called "Dining Philosophers" change this doc's title to something like "Dining Philosophers (CMSI RTOS V2 APIs)"

Author: @dbkinder: Done, thanks.

###################

Overview
********

> An implementation of a solution to the Dining Philosophers problem (a classic

Reviewer: Use a complete sentence, and lets add a reference to the "problem":
```suggestion
This sample implements a solution to the `Dining Philosophers problem
<https://en.wikipedia.org/wiki/Dining_philosophers_problem>`_ (a classic
```
